monitoring/memoryupdater.go

```diff
@@ -0,0 +1,85 @@
+package monitoring
+
+import (
+	"fmt"
+	"runtime"
+)
+
+const formatStr = "Name=%20s\t\tMax=%12v\t\tMin=%12v\t\tAvg=%12v\n"
+
+var names = []string{"NumGoroutine", "Alloc", "TotalAlloc", "Sys", "Mallocs", "Frees", "LiveObjects", "PauseTotalNs", "NumGC"}
+
+type formatter func(uint64) string
+
+func bytesToMBFormmater(x uint64) string {
+	return fmt.Sprintf("%vMB", x/1024/1024)
+}
+
+type MemoryUpdater struct {
+	memTracker   map[string]*Tracker
+	formatters   map[string]formatter
+	recordsCount int
+}
+
+func NewMemoryUpdater() *MemoryUpdater {
+	mu := new(MemoryUpdater)
+	mu.memTracker = make(map[string]*Tracker)
+	mu.formatters = make(map[string]formatter)
+	for _, name := range names {
+		mu.memTracker[name] = NewTracker()
+		mu.formatters[name] = nil
```

Comment:
why nil ?

Reply:
because the default is not to format the data, on special cases such as Alloc & TotalAlloc we use the bytesToMB formatter.